components/brave_wallet/common/solana_utils_unittest.cc

```diff
@@ -140,4 +140,46 @@ TEST(SolanaUtilsUnitTest, IsBase58EncodedSolanaPubkey) {
       Base58Encode(std::vector<uint8_t>(kSolanaPubkeySize - 1, 0))));
 }
 
+TEST(SolanaUtilsUnitTest, Uint8ArrayDecode) {
+  const struct {
+    const char* input;
+    const size_t len;
+    const std::vector<uint8_t> output;
+  } valid_cases[] = {{"[34,  208, 53,  54,  75,  46,  112, 55]", 8,
+                      std::vector<uint8_t>({34, 208, 53, 54, 75, 46, 112, 55})},
+                     {"[0, 0, 0, 0]", 4, std::vector<uint8_t>({0, 0, 0, 0})},
+                     {"[34,  208, 53,  54,  75,  46,  112, 55, 123, 15,  232, "
+                      "9,   45,  178, 252, 196, 62,  64,  169, 213, 66,  87,  "
+                      "192, 16, 152, 108, 254, 148, 183, 39,  51,  192]",
+                      kSolanaPubkeySize,
+                      std::vector<uint8_t>(
+                          {34,  208, 53,  54,  75,  46,  112, 55,  123, 15, 232,
+                           9,   45,  178, 252, 196, 62,  64,  169, 213, 66, 87,
+                           192, 16,  152, 108, 254, 148, 183, 39,  51,  192})}};
+
+  for (const auto& valid_case : valid_cases) {
+    std::vector<uint8_t> bytes;
+    EXPECT_TRUE(Uint8ArrayDecode(valid_case.input, &bytes, valid_case.len));
+    EXPECT_EQ(bytes, valid_case.output);
+  }
+
+  const struct {
+    const char* input;
+    const size_t len;
+  } invalid_cases[] = {{"[34}", 1},
```

Comment:
Worth also test for `"[]"`, `"[,]"`, `""`, `"["`, `"]"`

Reply:
added and squashed thx.